### PR TITLE
Forbid globals globally

### PR DESCRIPTION
**Pull request by jridgewell**
Also adds `global` to the globally forbidden globals.

This’ll prevent #10112, #8514, and likely others.

I didn't do this when I first forbade globals because it seemed like too much work. But we're getting too many bugs.

### CODE REVIEW

**Comment by aghassemi** [comment]
@jridgewell I don't think the issue is use of browser-globals, (`self` is a global anyway). I believe the problem is the non-browser-global variable `global` that Browserify(?) is defining and assigning `window` to, causing min/max to behave differently.

Anyway, we don't need to do all this, just removing `global` from `"globals": ` in `eslintrc` should catch these.

**Comment by aghassemi** [comment]
revert other changes? `window` is more descriptive than `self` IMO.

**Comment by jridgewell** [comment]
Reverted the `self`/`window` changes.

File: build-system/eslint-rules/no-global.js [hunk 1]
```diff
@@ -20,6 +20,12 @@ var astUtils = require('eslint/lib/ast-utils');
 var GLOBALS = Object.create(null);
 GLOBALS.window = 'Use `self` instead.';
 GLOBALS.document = 'Reference it as `self.document` or similar instead.';
+GLOBALS.global = "This ain't node."
```
Comment (aghassemi) [inline build-system/eslint-rules/no-global.js]: lol

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 2]
```diff
@@ -77,7 +77,7 @@ const DOUBLECLICK_BASE_URL =
 const PAGE_LEVEL_PARAMS_ = {
   'gdfp_req': '1',
   'sfv': DEFAULT_SAFEFRAME_VERSION,
-  'u_sd': window.devicePixelRatio,
+  'u_sd': self.devicePixelRatio,
```
Comment (aghassemi) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: window

File: extensions/amp-crypto-polyfill/0.1/amp-crypto-polyfill.js [hunk 3]
```diff
@@ -23,4 +23,4 @@ export function installCryptoPolyfill(win) {
   });
 }
 
-installCryptoPolyfill(window);
+installCryptoPolyfill(self);
```
Comment (aghassemi) [inline extensions/amp-crypto-polyfill/0.1/amp-crypto-polyfill.js]: window (and few more places in the PR still use `self` )

**Comment by jridgewell** [comment]
Everything under `extensions` needs to use `self` so I can lint for the other ones.

**Comment by aghassemi** [comment]
@jridgewell I am a bit confused now, wouldn't something like https://github.com/ampproject/amphtml/pull/10165 catche what caused #10112, #8514, etc? Why are the extra stuff here needed?

**Comment by jridgewell** [comment]
https://github.com/ampproject/amphtml/issues/2708, mainly for cross-iframe issues (testing). And https://github.com/ampproject/amphtml/pull/4531, but that's admittedly less of a concern for extension code.

**Comment by aghassemi** [comment]
@jridgewell fine to close this PR?

**Pull request closed without merging**